src/NCronJob/Scheduler/TaskScheduler/DeterministicTaskScheduler.cs

```diff
@@ -0,0 +1,45 @@
+namespace NCronJob;
+
+/// <summary>
+/// Deterministic and ordered execution: Use when the order of task execution is critical, and tasks may have dependencies that require strict ordering.
+/// </summary>
+internal class DeterministicTaskScheduler : TaskScheduler
```

Comment:
Wouldn't it be easier to use a `ConcurrentQueue` instead of manual locking and using a `LinkedList`?

Reply:
Yes, already made the update.